gobblin-runtime/src/main/java/gobblin/runtime/api/package-info.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+/**
+ * The Gobblin Execution Model.
+ *
+ * <p>The Gobblin launcher components:
```

Comment:
In a meeting, we decided to use the terminology "Gobblin Instance" instead of "Gobblin Application" with respect to a instantiation of Gobblin that runs a collection of jobs as those job are not necessarily related.

The interface itself will be deprecated as it looks very generic and does not define (in terms of APIs) what a Gobblin application actually is.

There are 4 implementations of this interface currently: AzkabanJobLauncher, CliLocalJobLauncher, CliMRJobLauncher, and ServiceBasedAppLauncher. The first three implement both ApplicationLauncher and JobLauncher which is kind of strange. Running a Gobblin Application (aka Instance) is more than just executing a job -- it is controlling the entire lifecycle of jobs. These three should focus on their primary responsibility -- executing a single job. The lifecycle management of Jobs should be left to the GobblinInstanceLauncher/GobblinInstanceDriver . We'll probably remove the "implements ApplicationLauncher" part (may keep the start/stop methods for backwards compatibility).

ServiceBasedAppLauncher will become part of a standard implementation of GobblinInstanceLauncher that sets up a standard environment for running Gobblin jobs.